components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1118,6 +1119,68 @@ public OAuthRevocationResponseDTO revokeAuthzForAppsByResourceOwner(
         return new OAuthRevocationResponseDTO();
     }
 
+    /**
+     * Revoke issued tokens for the application.
+     *
+     * @param consumerKey Consumer key for the application.
+     * @throws IdentityOAuthAdminException Error while revoking the issued tokens.
+     */
+    public void revokeIssuedTokensByConsumerKey(String consumerKey) throws IdentityOAuthAdminException {
+
+        int countToken = 0;
+        List<AccessTokenDO> activeDetailedTokens;
+        try {
+            //Get active access tokens by application id
+            activeDetailedTokens = new ArrayList<>(OAuthTokenPersistenceFactory
+                    .getInstance().getAccessTokenDAO().getActiveAcessTokenDataByConsumerKey(consumerKey));
+        } catch (IdentityOAuth2Exception e) {
+            String errorMsg = "Error occurred while retrieving access tokens issued for Client ID : " + consumerKey;
+            throw handleError(errorMsg, e);
+        }
+        triggerBulkPreRevokeListeners(consumerKey, activeDetailedTokens);
+        String[] accessTokens = new String[activeDetailedTokens.size()];
+        AuthenticatedUser authzUser;
+        for (AccessTokenDO detailToken : activeDetailedTokens) {
+            String token = detailToken.getAccessToken();
+            accessTokens[countToken] = token;
+            countToken++;
```

Comment:
Do the increment in the same line.

Reply:
Addressed in [18e795](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1627/commits/18e7958d172592cf2743ae36381a7fbecd2b74e0)